WooCommerce/Classes/POS/Models/CartItem+Order.swift

```diff
@@ -0,0 +1,37 @@
+import struct Yosemite.Order
+import struct Yosemite.OrderItem
+import protocol Yosemite.POSItem
+
+extension CartItem {
+    /// Checks if the order has the same items as the cart does.
+    /// Used in POS to check if we need to resync the Order with cart items
+    /// - Returns: Whether the order has the same items as the cart does
+    static func areOrderAndCartDifferent(order: Order?, cartItems: [CartItem]) -> Bool {
+        guard let order else {
+            return cartItems.isNotEmpty
+        }
+        // first we get list of all products 1 by 1
+        var singleQuantityOrderItems: [OrderItem] = order.items.flatMap {
+            Array(repeating: $0, count: $0.quantity.intValue)
+        }
+        var singleQuantityCartItems: [POSItem] = cartItems.flatMap {
+            Array(repeating: $0.item, count: $0.quantity)
+        }
+        // if order and cart do not have the same count of items
+        // we do not need to compare them one by one
+        if singleQuantityOrderItems.count == singleQuantityCartItems.count {
+            // sort items by productIDs to have them in same order for comparison
+            singleQuantityOrderItems.sort { $0.productID < $1.productID }
+            singleQuantityCartItems.sort { $0.productID < $1.productID }
+            // check if product item is different, prices included
+            for (index, cartItem) in singleQuantityCartItems.enumerated() {
+                let orderItem = singleQuantityOrderItems[index]
+                if cartItem.productID != orderItem.productID || cartItem.price != orderItem.price.stringValue {
```

Comment:
In my case, `cartItem.price` and `orderItem.price` are unequal. So this method always returns false and always syncs the order. Would it be enough to compare productIds? 

<img width="579" alt="image" src="https://github.com/user-attachments/assets/ee45bcfe-7f94-4647-ab4e-5de9a3e18abe">

A little concern about this method that it will need to expand and get more complicated once we support more complex products. One idea would be to save `cartItems` in memory when 'Checkout' is tapped. Next time it's tapped, we compare `previousCartItem` with the current ones. Although not sure if it wouldn't introduce some other issues 🤔

Reply:
Hm, good catch.. that is that issue with taxes being part of the price... I think I will update this a bit later and then we can look again :)

Reply:
I think I will update it to just check the IDs and add a todo comment for the future